### DISCUSSION TITLE
Queries regarding Reports available on site admin dashboard

### DISCUSSION THREAD

**Discussion by himankgupta81** [community/community#23709] [Enterprise]
<p>I have a few questions regarding the all_users.csv report available in the Site Admin dashboard:<br>
1- What is the criteria for defining a user dormant?<br>
2- Is there any time lag between the activity performed and the last active date updated in the report? If yes, then how much?<br>
3- Which all activities qualify for updating the last activity date of the user?<br>
4- is this report available only for the GitHub Enterprise on Premise or is it also available for <a href="http://github.com" rel="noopener nofollow ugc">github.com</a>?</p>

**Comment by francisfuzz** [comment, marked as answer]
<p>:wave: <a class="mention" href="/himankgupta81">@himankgupta81</a> , thanks for raising these questions!</p>
<blockquote>
<p>1- What is the criteria for defining a user dormant?</p>
</blockquote>
<p>For <a href="https://docs.github.com/en/enterprise-server@2.22/admin/configuration/site-admin-dashboard">GitHub Enterprise Server <code>2.22</code></a>, a user account is considered to be inactive (“dormant”) when it:</p>
<ul>
<li>Has existed for longer than the dormancy threshold that’s set for your GitHub Enterprise Server instance.</li>
<li>Has not generated any activity within that time period.</li>
<li>Is not a site administrator.</li>
</ul>
<p>Our team wrote more about the dormancy threshold in the <a href="https://docs.github.com/en/enterprise-server@2.22/admin/configuration/site-admin-dashboard#dormant-users"><code>Dormant users</code> section</a>, in addition to <a href="https://docs.github.com/en/enterprise-server@2.22/admin/guides/user-management/managing-dormant-users/#configuring-the-dormancy-threshold">configuring that threshold in this guide</a>.</p>
<p>If you’re using a different GitHub Enterprise Server version than <code>2.22</code>, there’s a dropdown showcasing all the other supported versions that you can view the content in:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/b/4b5d9888bd60a6be7a5ebcce0546ae770a5ae8f4.gif" data-download-href="/uploads/short-url/aKIkYXng0WU3J2ZUwt8s4ZMqqkA.gif?dl=1" title="ghes-version-change"><img src="https://user-images.githubusercontent.com/15894826/181078857-a6de566f-0966-467a-9a8f-a5cbd64826bf.gif" alt="ghes-version-change" data-base62-sha1="aKIkYXng0WU3J2ZUwt8s4ZMqqkA" width="690" height="381" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4b5d9888bd60a6be7a5ebcce0546ae770a5ae8f4_2_690x381.gif, https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4b5d9888bd60a6be7a5ebcce0546ae770a5ae8f4_2_1035x571.gif 1.5x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4b5d9888bd60a6be7a5ebcce0546ae770a5ae8f4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">ghes-version-change</span><span class="informations">1920×1062 2.01 MB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<blockquote>
<p>4- is this report available only for the GitHub Enterprise on Premise or is it also available for <a href="http://github.com/">github.com </a>?</p>
</blockquote>
<p>This report is available only for GitHub Enterprise Server (also known as GitHub Enterprise On Premise).</p>
<hr>
<blockquote>
<p>2- Is there any time lag between the activity performed and the last active date updated in the report? If yes, then how much?</p>
</blockquote>
<blockquote>
<p>3- Which all activities qualify for updating the last activity date of the user?</p>
</blockquote>
<p>I’ve asked my colleagues for an assist on these two questions and will write back as soon as I have an update. :+1:</p>

**Comment by francisfuzz** [comment]
<p>:wave:  <a class="mention" href="/himankgupta81">@himankgupta81</a>, I have an update! :smile_cat:</p>
<blockquote>
<p>2- Is there any time lag between the activity performed and the last active date updated in the report? If yes, then how much?</p>
</blockquote>
<p>I spoke with one of my colleagues and they shared their input with me:</p>
<blockquote>
<p><em>When a GitHub Enterprise Server site administrator requests the  <strong>All Users</strong>  Stafftools report, a background job is started to generate the report based on information contained in the GitHub Enterprise Server database. Once the background job completes the report is available for download. The time it takes for the Stafftools report background job to run may vary depending on the current load and performance of your GitHub Enterprise Server, although on non-busy instances we’d usually expect the job to completed within several minutes.</em></p>
</blockquote>
<blockquote>
<p><em>For a performant GitHub Enterprise Server instance, there shouldn’t be any delay between actions the user performs and the user’s  <code>last_active</code>  date as reflected in the Stafftools reports.</em></p>
</blockquote>
<hr>
<p>You also asked:</p>
<blockquote>
<p>3- Which all activities qualify for updating the last activity date of the user?</p>
</blockquote>
<p>Here’s a list of actions that will update the  <code>last_active</code>  field:</p>
<ul>
<li>
<strong>active sessions:</strong>  An active session is a day that the person has visited the site while logged in. Excludes people who have visited the same day they signed up.</li>
<li>
<strong>web edits:</strong>  A web edit is using the web UI to create a branch, tag, or do a file<br>
create, edit, delete operation.</li>
<li>
<strong>pull requests:</strong>  Submitting a new pull request</li>
<li>
<strong>issues:</strong>  Submitting a new issue.</li>
<li>
<strong>comments:</strong>  Commenting on an a Commit, an Issue, a Pull Request, and Pull Request Review Comments.</li>
<li>
<strong>pushes:</strong>  Git Pushes.</li>
<li>
<strong>wiki edits:</strong>  Creating, editing, and deleting a wiki page.</li>
<li>
<strong>releases:</strong>  Creating a new Release</li>
</ul>
<p>We hope this helps!</p>
